Dixit_Logic/Classes/DixitGame.cs

```diff
@@ -134,16 +135,20 @@ public void EvaluatePoints()
         }
 
         /// <summary>
-        /// If game is runing than this method add a new association text to the game state
+        /// If game is runing than this method add a new association text to the 
+        /// game state and set story teller (ActualPlayer) to the player who guess this text(in param).
         /// </summary>
         /// <param name="storyText">An association text string</param>
+        /// <param name="player">The player who guess the text</param>
         /// <returns>True if game's round status is in AssociationTelling phase and game
         /// is runing. Otherwise return false.  </returns>
-        public bool AddAssociationText(string storyText)
+        public bool AddAssociationText(string storyText, IPlayer player)
         {
-            if(_actGameState.GameIsRuning && _actGameState.RoundStatus == PhaseStatus.AssociationTelling)
+            if(_actGameState.RoundStatus == PhaseStatus.AssociationTelling)
```

Comment:
I think in this if clause we need to check if the player in the parameter is equal to the player who we are expecting to hear the story from.